src/Analyzer.Utilities/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -18,6 +19,53 @@ public static IEnumerable<INamedTypeSymbol> GetBaseTypesAndThis(this INamedTypeS
             }
         }
 
+        /// <summary>
+        /// Returns a value indicating whether <paramref name="type"/> derives from, or implements
+        /// any generic construction of, the type defined by <paramref name="parentType"/>.
+        /// </summary>
+        /// <remarks>
+        /// This method only works when <paramref name="parentType"/> is a definition,
+        /// not a constructed type.
+        /// </remarks>
+        /// <example>
+        /// <para>
+        /// If <paramref name="parentType"/> is the class <code>Stack&gt;T></code>, then this
+        /// method will return <code>true</code> when called on <code>Stack&gt;int></code>
+        /// or any type derived it, because <code>Stack&gt;int></code> is constructed from
+        /// <code>Stack&gt;T></code>.
+        /// </para>
+        /// <para>
+        /// Similarly, if <paramref name="parentType"/> is the interface <code>IList&gt;T></code>, 
+        /// then this method will return <code>true</code> for <code>List&gt;int></code>
+        /// or any other class that extends <code>IList&gt;></code> or an class that implements it,
+        /// because <code>IList&gt;int></code> is constructed from <code>IList&gt;T></code>.
+        /// </para>
+        /// </example>
+        public static bool DerivesFromOrImplementsAnyConstructionOf(this INamedTypeSymbol type, INamedTypeSymbol parentType, Compilation compilation)
+        {
+            if (!parentType.IsDefinition)
+            {
+                throw new ArgumentException($"The type {nameof(parentType)} is not a definition; it is a constructed type", nameof(parentType));
```

Comment:
Per discussion: We don't localize exception strings in Roslyn, so un-localizing the one in `StringExtensions`.